### DISCUSSION TITLE
execute npx commands with import alias context

### DISCUSSION THREAD

**Discussion by kelbyfaessler** [sveltejs/kit#12148] [Q&A]
There have been several discussion questions asking how to execute custom scripts (e.g. myscript.ts) where the scripts need to be aware of sveltekits import aliases (like $lib). The answer has been to use `vite-node` to execute scripts, which works for me.

What about 3rd party scripts executed via npx? For example, if I'm using drizzle the command to generate migrations is 
`npx drizzle-kit generate:sqlite`

but if I use sveltekit's import aliases in any files that drizzle reads, it fails.

I've tried various ways to using vite-node to execute this command but so far nothing has worked. Has anyone else solved this problem?

**Comment by MikeSpock** [comment]
Facing the same issue, would appreciate it if someone had an answer.

A dirty workaround I found is to run utility functions in vitest files. Vitest does create the alias import contexts, and you can config it to also run on *.utility.js files. 
then you may add the cli command to your package.json that only runs that single utility file, or just a single test clause from that file.

Anyone found a good solution to this?

**Comment by david-plugge** [comment]
https://github.com/sveltejs/kit/discussions/9807#discussioncomment-5792202

**Reply by kelbyfaessler** [reply]
Reread the description here. This is a different use case where that solution doesn't work.

> There have been several discussion questions asking how to execute custom scripts (e.g. myscript.ts) where the scripts need to > be aware of sveltekits import aliases (like $lib). The answer has been to use vite-node to execute scripts, which works for me.
> 
> What about 3rd party scripts executed via npx? For example, if I'm using drizzle the command to generate migrations is
> npx drizzle-kit generate:sqlite

**Reply by david-plugge** [reply]
In that case you can use drizzle kit programmatically. As sveltekit is built in top of vite and all the features like special imports are vite plugins there is no general solution

**Reply by kelbyfaessler** [reply]
What do you mean by `use drizzle kit programmatically`?

Drizzle's recommended way of generating migration files is to run `npx drizzle-kit generate:sqlite` . The question is how do I run this command, when some of the files it uses have sveltekit import syntax that won't allow it to be run in this way?
